parser-typechecker/src/Unison/Lexer.hs

```diff
@@ -696,23 +704,36 @@ keywords = Set.fromList [
   "where", "use",
   "true", "false",
   "type", "ability", "alias", "typeLink", "termLink",
-  "let", "namespace", "case", "of"]
+  "let", "namespace", "match"]
 
 -- These keywords introduce a layout block
 layoutKeywords :: Set String
 layoutKeywords =
   Set.fromList [
-    "if", "handle", "let", "where", "of"
+    "if", "handle", "let", "where", "match"
   ]
 
 -- These keywords end a layout block and begin another layout block
 layoutCloseAndOpenKeywords :: Set String
 layoutCloseAndOpenKeywords = Set.fromList ["then", "else", "with"]
 
-openingKeyword :: String -> String
-openingKeyword "then" = "if"
-openingKeyword "else" = "then"
-openingKeyword "with" = "handle"
+-- Use a transformed block name to disambiguate certain keywords
+layoutCloseAndOpenKeywordMap :: String    -- close-and-open keyword
+                             -> Layout    -- layout
+                             -> BlockName -- transformed blockname for keyword
+layoutCloseAndOpenKeywordMap "with" l =
+  case findNearest l (Set.fromList ["handle", "match"]) of
+    Just "match"  -> "match-with"
+    Just "handle" -> "handle-with"
+    _ -> "with"
+layoutCloseAndOpenKeywordMap kw _ = kw
+
+openingKeyword :: BlockName -> String
+openingKeyword        "then" = "if"
+openingKeyword        "else" = "then"
+openingKeyword        "with" = "match or handle" -- hack!!
```

Comment:
This is used only when `with` appears without a matching `match` or `handle` first, which is an error.  The "match or handle" string winds up in the error message, colored red.

Reply:
Epic hack! It's fine though. In general, I'd put this info in a comment in the code rather than the PR notes.